### ISSUE TITLE
nembed not calculated correctly when nullptr passed

### ISSUE THREAD

**Issue by bd4** [ROCmSoftwarePlatform/hipFFT#3]
### What is the expected behavior
-  Passing nullptr to inembed and onembed to hipfftPlanMany, with Z2D, will calculate the correct values for typical case (no padding)

### What actually happens
-  Code runs without errors, but execute result contains values that are clearly wrong

### How to reproduce
-  Pass nullptr to hipfftPlanMany for both inembed and onembed, and use Z2D or C2R type (forward FFTs may have same issue). See (https://github.com/bd4/fft-examples/blob/main/fft2d_hip.cxx#L120) for complete example. Corresponding cuFFT code works fine.

### Environment
| Hardware | description |
|-----|-----|
| GPU | gfx803 |
| CPU | Intel(R) Core(TM) i7-8700K |

| Software | version |
|-----|-----|
| ROCm | v4.1.0 (ubuntu 20.04 packages) |

**Comment by bd4** [comment]
Workaround is to specify them explicitly:
```
    int n[2] = { Nx, Ny };
    int inembed[2] = { n[0], n[1] / 2 + 1 };
    rc = hipfftPlanMany(&plan2, 2, n,
                        inembed, 1, Nx*Nycomplex,
                        n, 1, Nx*Ny,
                        HIPFFT_Z2D, 1);
```

**Comment by malcolmroberts** [comment]
Thanks for raising this issue.  Is this an in-place transform that you tried?

**Comment by bd4** [comment]
Out of place, did not try in-place.

**Comment by evetsso** [comment]
It looks like this was fixed by f129e610fa496fd5c343aa7094e47848c4f6cca4 which is not present in ROCm 4.1, but is present in develop and will be present in ROCm 4.2.  When I run your example against a hipFFT built from develop, the results look good.

Since it sounds like you already have a workaround for the problem, I'll close this issue.  Please reopen if you still have issues.
